pkg/manifests/amcfg.go

```diff
@@ -180,6 +185,30 @@ func ConvertToThanosAlertmanagerConfiguration(ta []AdditionalAlertmanagerConfig)
 
 		cfg.StaticConfigs = a.StaticConfigs
 
+		httpConfig := httpproxy.Config{
+			HTTPProxy:  f.proxy.HTTPProxy(),
+			HTTPSProxy: f.proxy.HTTPSProxy(),
+			NoProxy:    f.proxy.NoProxy(),
+		}
+
+		proxyFunc := httpConfig.ProxyFunc()
+
+		if len(cfg.StaticConfigs) > 0 && cfg.StaticConfigs[0] != "" {
```

Comment:
It would be good to have a comment here explaining that we assume all hosts follow the same proxy policy (e.g. use a proxy or none). As well as a note in the configuration documentation to avoid user misunderstanding.

Reply:
(nit) we could also iterate over all hosts and stop at the first one returning a non-nil proxy URL. I feel that it's more predictable and simpler to document, WDYT?

Reply:
I did it like that because proxyFunc only return nil if no proxy is defined, then the scenario of a host returning nil is not possible. 

From api:
`// A nil URL and nil error are returned if no proxy is defined in the
// environment, or a proxy should not be used for the given request, as
// defined by NO_PROXY.
//
// As a special case, if req.URL.Host is "localhost" or a loopback address
// (with or without a port number), then a nil URL and nil error will be returned.

But your way it is better for the documentation

Reply:
done